vars/pipeline2ATX.groovy

```diff
@@ -266,23 +266,41 @@ def calculateTime(executionTestSteps, build) {
         }
     }
 
-    int totalDuration = queueDuration + setupDuration + executionDuration + teardownDuration
+    def totalDuration = queueDuration + setupDuration + executionDuration + teardownDuration
 
-    int setupPercentage = (setupDuration / totalDuration) * 100
-    int executionPercentage = (executionDuration / totalDuration) * 100
-    int teardownPercentage = (teardownDuration / totalDuration) * 100
+    def setupPercentage = 0.0
+    def executionPercentage = 0.0
+    def teardownPercentage = 0.0
+
+    if (totalDuration > 0) {
+        setupPercentage = (setupDuration / totalDuration) * 100
+        executionPercentage = (executionDuration / totalDuration) * 100
+        teardownPercentage = (teardownDuration / totalDuration) * 100
+    }
 
     return [
-            setupDuration: setupDuration,
-            setupPercentage: setupPercentage,
-            executionDuration: executionDuration,
-            executionPercentage: executionPercentage,
-            teardownDuration: teardownDuration,
-            teardownPercentage: teardownPercentage,
-            queueDuration: queueDuration,
-            totalDuration: totalDuration,
-            fromCommitToStartTime: fromCommitToStartTime != null ? (int) fromCommitToStartTime : null,
-            errorTime: errorTime != null ? (int) (setupDuration + queueDuration + errorTime) : null]
+            setupDuration: convertTimeValueToDouble(setupDuration),
+            setupPercentage: convertTimeValueToDouble(setupPercentage),
+            executionDuration: convertTimeValueToDouble(executionDuration),
+            executionPercentage: convertTimeValueToDouble(executionPercentage),
+            teardownDuration: convertTimeValueToDouble(teardownDuration),
+            teardownPercentage: convertTimeValueToDouble(teardownPercentage),
+            queueDuration: convertTimeValueToDouble(queueDuration),
+            totalDuration: convertTimeValueToDouble(totalDuration),
+            fromCommitToStartTime: fromCommitToStartTime != null ? convertTimeValueToDouble(fromCommitToStartTime) : null,
+            errorTime: errorTime != null ? convertTimeValueToDouble(setupDuration + queueDuration + errorTime) : null]
+}
+
+/**
+ * Converts a none null time value to a double with one decimal place.
+ * @param value
+ *      the time value to be converted
+ * @return the time value converted as double
+ */
+def convertTimeValueToDouble(def value) {
+    double doubleValue = value as double
+    String formatted = String.format("%.1f", doubleValue).replace(',', '.')
```

Comment:
you easily could use 
`return String.format("%.1f", value).toDouble()` or `return (Math.round(value* 10.0)/10.0).toDouble()`
Since test.guide support strings as well (see schema file) i would prefer the first solution to avoid any additional parsing